### PR TITLE
[SPARK-31784][CORE][TEST] Fix test BarrierTaskContextSuite."share messages with allGather() call"

### PR DESCRIPTION
**Pull request by Ngone51**
### What changes were proposed in this pull request?

Change from `messages.toList.iterator` to `Iterator.single(messages.toList)`.

### Why are the changes needed?

In this test, the expected result of `rdd2.collect().head` should actually be `List("0", "1", "2", "3")` but is `"0"` now.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Updated test.

Thanks @WeichenXu123 reported this problem.

### CODE REVIEW

**Comment by Ngone51** [comment]
ping @sarthfrey @WeichenXu123 @jiangxb1987

**Review by jiangxb1987 — APPROVED** [review]
LGTM

**Comment by jiangxb1987** [comment]
retest this please

**Comment by Ngone51** [comment]
retest this please

File: core/src/test/scala/org/apache/spark/scheduler/BarrierTaskContextSuite.scala [hunk 1]
```diff
@@ -69,12 +69,12 @@ class BarrierTaskContextSuite extends SparkFunSuite with LocalSparkContext with
...
       val message: String = context.partitionId().toString
       val messages: Array[String] = context.allGather(message)
-      messages.toList.iterator
+      Iterator.single(messages.toList)
     }
     // Take a sorted list of all the partitionId messages
     val messages = rdd2.collect().head
     // All the task partitionIds are shared
-    for((x, i) <- messages.view.zipWithIndex) assert(x.toString == i.toString)
+    assert(messages === List("0", "1", "2", "3"))
```
Comment (WeichenXu123) [inline core/src/test/scala/org/apache/spark/scheduler/BarrierTaskContextSuite.scala:77]: We'd better verify the whole rdd collect array, not only the head of result array?

Comment (Ngone51) [reply core/src/test/scala/org/apache/spark/scheduler/BarrierTaskContextSuite.scala:77]: make sense.

**Comment by cloud-fan** [comment]
LGTM

**Comment by jiangxb1987** [comment]
Thanks, merged to master/3.0 !

**Comment by Ngone51** [comment]
thanks all!

**Pull request closed without merging**